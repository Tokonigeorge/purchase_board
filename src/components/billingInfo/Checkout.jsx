import PaymentButton from "../PaymentButton";

const Checkout = ({ checkoutData, handleNav, handleCancel }) => {
  return (
    <>
      <div className="w-full bg-white rounded-md my-12 pb-8 shadow-lg">
        <div className="flex justify-between items-center px-12 py-3.5 bg-blueMain rounded-t-md text-white text-sm font-semi-bold">
          <p>Item Name</p>
          <p>&#8358; Price</p>
        </div>
        <div className="mt-6 text-purp text-sm font-light pb-6 mx-6 px-6 border-b border-lightBlue">
          {checkoutData?.map((i, indx) => {
            return (
              <div
                className={`flex justify-between items-center pt-7 `}
                key={i.item + indx}
              >
                <p>{i.item}</p>
                <p
                  className={`${
                    i.item === "Shipping" && "text-black font-bold"
                  }`}
                >
                  {i.value}
                </p>
              </div>
            );
          })}
        </div>
        <div className="flex justify-between px-3 mx-9 py-2 ring-1 mt-8 rounded-sm ring-lightBlue items-center text-sm">
          <p className="text-lightBlue">Total</p>
          <p className="font-bold text-purp">
            {checkoutData
              .map((i) => Number(i.value.replace(/,/g, "")))
              .reduce((a, b) => a + b, 0)
              .toFixed(2)}
          </p>
        </div>
      </div>
      <PaymentButton
        handleNav={handleNav}
        handleCancel={handleCancel}
        text="Pay"
      />
    </>
  );
};

export default Checkout;
