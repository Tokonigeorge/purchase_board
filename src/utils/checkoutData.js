export const checkoutData = [
  {
    item: "Data science and usability",
    value: "50,000.00",
  },
  {
    item: "Shipping",
    value: "0.00",
  },
];
