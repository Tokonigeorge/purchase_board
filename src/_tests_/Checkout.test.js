/**
 * @jest-environment jsdom
 */
import * as React from "react";

import renderer from "react-test-renderer";
import Checkout from "../components/billingInfo/Checkout";
import { checkoutData } from "../utils/checkoutData";

describe("It renders Sucessfully", () => {
  test("eventHandler called on click", () => {
    const tree = renderer.create(<Checkout checkoutData={checkoutData} />);
    //create a snapshot for the component
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
