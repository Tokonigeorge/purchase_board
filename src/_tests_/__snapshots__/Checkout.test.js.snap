// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Sucessfully eventHandler called on click 1`] = `
Array [
  <div
    className="w-full bg-white rounded-md my-12 pb-8 shadow-lg"
  >
    <div
      className="flex justify-between items-center px-12 py-3.5 bg-blueMain rounded-t-md text-white text-sm font-semi-bold"
    >
      <p>
        Item Name
      </p>
      <p>
        â‚¦ Price
      </p>
    </div>
    <div
      className="mt-6 text-purp text-sm font-light pb-6 mx-6 px-6 border-b border-lightBlue"
    >
      <div
        className="flex justify-between items-center pt-7 "
      >
        <p>
          Data science and usability
        </p>
        <p
          className="false"
        >
          50,000.00
        </p>
      </div>
      <div
        className="flex justify-between items-center pt-7 "
      >
        <p>
          Shipping
        </p>
        <p
          className="text-black font-bold"
        >
          0.00
        </p>
      </div>
    </div>
    <div
      className="flex justify-between px-3 mx-9 py-2 ring-1 mt-8 rounded-sm ring-lightBlue items-center text-sm"
    >
      <p
        className="text-lightBlue"
      >
        Total
      </p>
      <p
        className="font-bold text-purp"
      >
        50000.00
      </p>
    </div>
  </div>,
  <div
    className="flex items-center gap-x-6 pb-24 mt-8"
  >
    <button
      className="text-white bg-gradient-to-r from-btnGradOne to-btnGradTwo py-2.5 rounded-lg px-16 font-bold text-sm "
      data-testid="test-btn"
      onClick={[Function]}
    >
      Pay
    </button>
    <button
      className="text-purp pl-5 font-bold text-sm "
      data-testid="test-btn"
      onClick={[Function]}
    >
      Cancel Payment
    </button>
  </div>,
]
`;
